portal-bridge-nodes.md

```diff
@@ -0,0 +1,37 @@
+# Bridge Node JSON-RPC Interface
+
+### Original Link: https://notes.ethereum.org/@lithp/r1NB6M4Gu
+### Author: @lithp
+### Created: 2021-02-24
+
+# Description
+Bridge nodes push state into the state availability network. They need to be in communication with an eth1 full node, here is a protocol they could use.
+
+### Bridge nodes will send two kinds of messages:
+- (ingress) an account/slot has some value
+	- This is useful for adding new accounts/slots to the network, updating existing accounts/slots, as well as for re-inserting cold accounts/slots which the network might have forgotten
+- (egress) an account/slot has been removed
+	- The state availability network is not an archive node, it only stores the last x⪅1000 state tries. A recent proof of non-inclusion is enough to convince the network to drop data.
+	- There is some logic to selectively ignore these messages. If the account is brand new it’s possible to construct a recent non-inclusion proof but that proof is not relevant.
+
+# API
+### To do this they will rely on a few different RPCs:
+- `bridge_waitNewCanonicalChain()`
+	- Blocks the caller until the canonical chain tip changes (indicating there is more work to be done).
+
+- `bridge_getBlockChanges(blockHash) -> List[(Address, [Address, ...])]`
```

Comment:
The innermost `Address` in the return type looks wrong here. To conform with the types in `bridge_getNextItem`, I believe the return type should look like: `List[(acctAddr, [slotAddr, ...])]`

Reply:
Ahh, good catch - updated in #56